core/src/main/java/io/questdb/cairo/mv/MatViewGraph.java

```diff
@@ -145,6 +150,44 @@ private MatViewDependencyList getOrCreateDependentViews(CharSequence baseTableNa
         return dependentViewsByTableName.computeIfAbsent(baseTableName, createDependencyList);
     }
 
+    public boolean hasDependencyLoop(CharSequence baseTableName, TableToken newMatViewToken) {
+        CharSequenceHashSet seen = tlSeen.get();
+        ArrayDeque<CharSequence> stack = tlStack.get();
+
+        seen.clear();
+        stack.clear();
+
+        if (Chars.equals(baseTableName, newMatViewToken.getTableName())) {
```

Comment:
We need to use `Chars.equalsIgnoreCase` here and in the below check as table names are case-insensitive.

Reply:
fixed: https://github.com/questdb/questdb/pull/5592/commits/234015ce93f5c53cbba664ab30b0d081683c881d